main.py

```diff
@@ -1411,6 +1415,43 @@ def handle_response(self, response, pos):
                               {'changes': response})
 
 
+class LspFormatDocumentRangeCommand(sublime_plugin.TextCommand):
+    def is_enabled(self):
+        if is_supported_view(self.view):
+            client = client_for_view(self.view)
+            if client and client.has_capability('documentRangeFormattingProvider'):
+                for region in self.view.sel():
+                    # Let's check if there is at least one non-trivial region.
+                    if region.begin() != region.end():
+                        return True
+                return False
+        return False
+
+    def run(self, _):
+        self.client = client_for_view(self.view)
+        # Ask for range formatting for each selected region.
+        for region in sorted(self.view.sel(), key=lambda reg: -reg.begin()):
+            self._send_request(region)
+
+    def _send_request(self, region: sublime.Region) -> None:
+        debug("sending format range request for", region)
+        params = {
+            "textDocument": {
+                "uri": filename_to_uri(self.view.file_name())
+            },
+            "range": Range.from_region(self.view, region).to_lsp(),
+            "options": {
+                "tabSize": 4,  # TODO: Fetch these from the project settings / global settings
+                "insertSpaces": True
+            }
+        }
+        if not self.client:
+            return
```

Comment:
I couldn't manage to silence mypy without this check, but it is essentially redundant, because it is checked for in the `is_enabled` method. How can I tell mypy that?